addons/mass_mailing/static/src/js/mailing_m2o_filter.js

```diff
@@ -0,0 +1,158 @@
+/** @odoo-module **/
+
+import { registry } from '@web/core/registry';
+import { Many2OneField } from '@web/views/fields/many2one/many2one_field';
+import { useService } from "@web/core/utils/hooks";
+import Domain from 'web.Domain';
+
+const { onMounted, onPatched, useExternalListener } = owl;
+
+/**
+ * Widget to create / remove favorite filters on mass mailing and/or marketing automation, extended
+ * from Many2OneField. This widget is designed specifically for 'mailing_filter_id'
+ * field on 'mailing.mailing' and 'marketing.campaign' form view.
+ *
+ * In edit mode, it will allow to save the latest configured domain
+ * in form of the favorite filter, or to remove the store filters.
+ *
+ */
+export class FieldMany2OneMailingFilter extends Many2OneField {
+    setup() {
+        super.setup();
+        this.notification = useService("notification");
+        onMounted(this._updateFilterIcons);
+        onPatched(this._updateFilterIcons);
+        useExternalListener(document, "show.bs.dropdown", this.onFilterDropdownShown);
```

Comment:
That seems risky.

If any other dopdown is shown, then you will apply your behavior?

Can't we make a small Component out of this dropdown and handle events manually in there?